File: app/api/sms/route.ts
import { NextResponse } from "next/server"

// This would be replaced with actual Africa's Talking SDK in a real implementation
// import AfricasTalking from 'africastalking';

// Mock data for demonstration
const learningModules = [
  {
    id: "biz101",
    title: "Business Basics",
    lessons: [
      "Lesson 1: Identifying Market Opportunities",
      "Lesson 2: Creating a Business Plan",
      "Lesson 3: Understanding Your Target Market",
    ],
  },
  {
    id: "fin101",
    title: "Financial Literacy",
    lessons: [
      "Lesson 1: Bookkeeping Fundamentals",
      "Lesson 2: Pricing Your Products/Services",
      "Lesson 3: Managing Cash Flow",
    ],
  },
  {
    id: "mkt101",
    title: "Marketing Essentials",
    lessons: [
      "Lesson 1: Building Your Brand",
      "Lesson 2: Low-Cost Marketing Strategies",
      "Lesson 3: Using Social Media for Business",
    ],
  },
]

export async function POST(request: Request) {
  try {
    const data = await request.json()
    const { phoneNumber, message } = data

    if (!phoneNumber || !message) {
      return NextResponse.json({ error: "Phone number and message are required" }, { status: 400 })
    }

    // Process the incoming SMS
    const response = processSmsRequest(phoneNumber, message)

    // In a real implementation, this would use Africa's Talking SDK to send SMS
    // const response = await sendSmsResponse(phoneNumber, responseMessage);

    return NextResponse.json({ success: true, response })
  } catch (error) {
    console.error("SMS processing error:", error)
    return NextResponse.json({ error: "Failed to process SMS request" }, { status: 500 })
  }
}

function processSmsRequest(phoneNumber: string, message: string) {
  // Convert message to lowercase for easier processing
  const lowerMessage = message.toLowerCase().trim()

  // Handle registration
  if (lowerMessage.startsWith("register")) {
    return {
      type: "registration",
      message:
        "Welcome to SkillConnect! You have successfully registered. Reply with MENU to see available options or HELP for assistance.",
      phoneNumber,
    }
  }

  // Handle menu request
  if (lowerMessage === "menu") {
    return {
      type: "menu",
      message:
        "SkillConnect Menu:\n1. LEARN - Access learning modules\n2. ASSESS - Take a skill assessment\n3. MENTOR - Get matched with a mentor\n4. EVENTS - Upcoming workshops\n5. HELP - Get assistance",
      phoneNumber,
    }
  }

  // Handle learning module request
  if (lowerMessage === "learn") {
    const moduleList = learningModules.map((module, index) => `${index + 1}. ${module.title}`).join("\n")

    return {
      type: "learning",
      message: `Available Learning Modules:\n${moduleList}\nReply with the module number to access lessons.`,
      phoneNumber,
    }
  }

  // Handle specific module selection (e.g., "1" for Business Basics)
  if (/^[1-3]$/.test(lowerMessage)) {
    const moduleIndex = Number.parseInt(lowerMessage) - 1
    if (moduleIndex >= 0 && moduleIndex < learningModules.length) {
      const module = learningModules[moduleIndex]
      const lessonList = module.lessons.join("\n")

      return {
        type: "module",
        message: `${module.title}:\n${lessonList}\nReply with ${module.id} followed by lesson number (e.g., ${module.id} 1) to access a specific lesson.`,
        phoneNumber,
      }
    }
  }

  // Handle specific lesson request (e.g., "biz101 1")
  const lessonMatch = lowerMessage.match(/^([a-z]+\d+)\s+(\d+)$/)
  if (lessonMatch) {
    const moduleId = lessonMatch[1]
    const lessonNumber = Number.parseInt(lessonMatch[2])

    const module = learningModules.find((m) => m.id === moduleId)
    if (module && lessonNumber > 0 && lessonNumber <= module.lessons.length) {
      // In a real implementation, this would fetch the actual lesson content
      return {
        type: "lesson",
        message: `${module.lessons[lessonNumber - 1]}\n\nThis lesson covers essential concepts for entrepreneurs. We'll send you part 1 of 3 now:\n\nEntrepreneurship begins with identifying problems that need solutions. Look around your community - what challenges do people face? Each challenge is a potential business opportunity.`,
        phoneNumber,
      }
    }
  }

  // Handle help request
  if (lowerMessage === "help") {
    return {
      type: "help",
      message:
        "SkillConnect Help:\nREGISTER - Create an account\nMENU - View all options\nLEARN - Access learning modules\nASSESS - Take skill assessment\nMENTOR - Get mentor matching\nEVENTS - View upcoming events",
      phoneNumber,
    }
  }

  // Default response for unrecognized commands
  return {
    type: "unknown",
    message: "Command not recognized. Reply with MENU to see available options or HELP for assistance.",
    phoneNumber,
  }
}

// This function would use Africa's Talking SDK in a real implementation
// async function sendSmsResponse(phoneNumber: string, message: string) {
//   // Initialize the SDK
//   const africastalking = AfricasTalking({
//     apiKey: process.env.AT_API_KEY,
//     username: process.env.AT_USERNAME
//   });
//
//   // Get the SMS service
//   const sms = africastalking.SMS;
//
//   // Send the message
//   const result = await sms.send({
//     to: phoneNumber,
//     message: message,
//     from: 'SkillConnect'
//   });
//
//   return result;
// }

File: app/api/ussd/route.py
import { NextResponse } from "next/server"

// Mock data for demonstration
const skillAssessments = [
  {
    id: "business",
    title: "Business Planning Assessment",
    description: "Evaluate your business planning skills and knowledge.",
  },
  {
    id: "financial",
    title: "Financial Literacy Assessment",
    description: "Test your understanding of business finances and accounting.",
  },
  {
    id: "marketing",
    title: "Marketing Skills Assessment",
    description: "Assess your marketing and customer acquisition knowledge.",
  },
]

export async function POST(request: Request) {
  try {
    const data = await request.json()
    const { sessionId, phoneNumber, text, serviceCode } = data

    if (!sessionId || !phoneNumber || !serviceCode) {
      return NextResponse.json({ error: "Session ID, phone number, and service code are required" }, { status: 400 })
    }

    // Process the USSD request
    const response = processUssdRequest(sessionId, phoneNumber, text, serviceCode)

    return NextResponse.json({ success: true, response })
  } catch (error) {
    console.error("USSD processing error:", error)
    return NextResponse.json({ error: "Failed to process USSD request" }, { status: 500 })
  }
}

function processUssdRequest(sessionId: string, phoneNumber: string, text: string, serviceCode: string) {
  // USSD responses should be prefixed with "CON" for expecting more input
  // or "END" for ending the session

  // Initial menu (no text input yet)
  if (!text) {
    return {
      response: `CON Welcome to SkillConnect!
      
1. Take Skill Assessment
2. View Learning Progress
3. Find a Mentor
4. Access Resources
5. My Profile`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Handle skill assessment menu
  if (text === "1") {
    const assessmentList = skillAssessments.map((assessment, index) => `${index + 1}. ${assessment.title}`).join("\n")

    return {
      response: `CON Select an assessment:
      
${assessmentList}`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Handle specific assessment selection
  if (text.startsWith("1*")) {
    const selection = text.split("*")[1]
    const assessmentIndex = Number.parseInt(selection) - 1

    if (assessmentIndex >= 0 && assessmentIndex < skillAssessments.length) {
      const assessment = skillAssessments[assessmentIndex]

      return {
        response: `CON ${assessment.title}
        
${assessment.description}

1. Start Assessment
2. View Sample Questions
3. Back to Main Menu`,
        sessionId,
        phoneNumber,
        serviceCode,
      }
    }
  }

  // Handle starting an assessment
  if (text.match(/^1\*[1-3]\*1$/)) {
    const parts = text.split("*")
    const assessmentIndex = Number.parseInt(parts[1]) - 1
    const assessment = skillAssessments[assessmentIndex]

    return {
      response: `END You have started the ${assessment.title}.
      
We will send you the first question via SMS shortly.
      
Thank you for using SkillConnect!`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Handle viewing learning progress
  if (text === "2") {
    return {
      response: `CON Your Learning Progress:
      
Business Basics: 60% complete
Financial Literacy: 30% complete
Marketing Essentials: 45% complete

1. Continue Learning
2. View Certificates
3. Back to Main Menu`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Handle mentor finding
  if (text === "3") {
    return {
      response: `CON Find a Mentor:
      
1. Search by Business Sector
2. Search by Location
3. View Recommended Mentors
4. Back to Main Menu`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Handle resources
  if (text === "4") {
    return {
      response: `CON Access Resources:
      
1. Business Templates
2. Funding Opportunities
3. Local Business Networks
4. Success Stories
5. Back to Main Menu`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Handle profile
  if (text === "5") {
    return {
      response: `CON My Profile:
      
Phone: ${phoneNumber}
Membership: Basic
Assessments Completed: 2
Mentor Sessions: 1

1. Update Profile
2. Upgrade Membership
3. Back to Main Menu`,
      sessionId,
      phoneNumber,
      serviceCode,
    }
  }

  // Default response for unrecognized inputs
  return {
    response: `END Invalid selection. Please dial the service code again and try once more.`,
    sessionId,
    phoneNumber,
    serviceCode,
  }
}

